tests/benchmark/utils/s2n_server_recv_benchmark.cc

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tests/benchmark/utils/s2n_server_recv_benchmark.h"
+#include "tests/benchmark/utils/shared_info.h"
+#include <unistd.h>
+#include <stdio.h>
+#include <vector>
+#include <stdlib.h>
+#include <poll.h>
+
+extern "C" {
+#include "bin/common.h"
+#include "tls/s2n_connection.h"
+}
+
+int recv_data(struct s2n_connection *conn, int sockfd2, benchmark::State& state, bool warmup) {
+    struct pollfd readers[1];
+    readers[0].fd = sockfd2;
+    readers[0].events = POLLIN;
+    errno = 0;
+    int poll_output = 0;
+    int number_bytes_sending = state.range(0);
+    s2n_blocked_status blocked;
+    uint8_t* recieved;
+    recieved = (uint8_t*)malloc(number_bytes_sending);
+
+    poll_output = poll(readers, 1, -1);
+    int bytes_read = 0;
+    do {
+        int received_bytes = 0;
+        if (!warmup) {
+            state.ResumeTiming();
+        }
+        benchmark::DoNotOptimize(received_bytes = s2n_recv(conn, recieved + bytes_read, number_bytes_sending - bytes_read, &blocked));
+        if (!warmup) {
+            state.PauseTiming();
+        }
+        if (received_bytes < 0) {
+            fprintf(stderr, "Error reading from connection: '%s' %d\n", s2n_strerror(s2n_errno, "EN"),
+                    s2n_connection_get_alert(conn));
+            exit(1);
+        }
+        bytes_read += received_bytes;
+    } while (bytes_read < number_bytes_sending);
```

Comment:
Does it makes sense to move the start/pausing outside the loop? The customer will need to call this multiple times and it captures the total time to send/recv the data. Also I'm curious what kind of impact the start/pause has on the overall benchmark based on the warning in google benchmark's code.